intranet/apps/dataimport/management/commands/import_eighth.py

```diff
@@ -0,0 +1,63 @@
+import json
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand, CommandError
+from django.db.models import Q
+
+from intranet.apps.eighth.models import EighthActivity, EighthRoom, EighthSponsor
+
+
+class Command(BaseCommand):
+    help = "Import Fake Activities"
+
+    def add_arguments(self, parser):
+        # Positional arguments
+        parser.add_argument("data_fname")
+
+    def handle(self, *args, **kwargs):
+        try:
+            with open(kwargs["data_fname"]) as f_obj:
+                data = json.load(f_obj)
+        except OSError as ex:
+            raise CommandError(str(ex)) from ex
+
+        for activity in data["activities"]:
+            name = activity["Name"].strip()
+            description = activity["Description"].strip()
+            sponsor_name = activity["Sponsor"].strip().split()
+            room_num = activity["Room Number"]
+            capacity = activity["Capacity"]
+            date = activity["Date"].strip()
+            classification = activity["Classification"]
+            if len(sponsor_name[1]) > 7:
+                sponsor_username = sponsor_name[0][:1] + sponsor_name[1][:7]
+            else:
+                sponsor_username = sponsor_name[0][:1] + sponsor_name[1][:7]
+            if not get_user_model().objects.filter(Q(username=sponsor_username)).exists():
+                sponsor_object = get_user_model().objects.create(
+                    username=sponsor_username, first_name=sponsor_name[0], last_name=sponsor_name[1], user_type="teacher"
+                )
+            else:
+                sponsor_object = get_user_model().objects.get(username=sponsor_username)
+            sponsor, idc = EighthSponsor.objects.get_or_create(
+                first_name=sponsor_object.first_name, last_name=sponsor_object.last_name, user=sponsor_object
+            )
+            room, idc = EighthRoom.objects.get_or_create(name=room_num, capacity=capacity)
+            activity, idc = EighthActivity.objects.get_or_create(
+                name=name,
+                description=description,
+                default_capacity=capacity,
+                presign=(classification == "presign"),
+                one_a_day=(classification == "oneaday"),
+                sticky=(classification == "sticky"),
+                special=(classification == "special"),
+                administrative=(classification == "administrative"),
+                restricted=(classification == "restricted"),
+                wed_a=(date == "wed a"),
+                wed_b=(date == "wed b"),
+                fri_a=(date == "fri a"),
+                fri_b=(date == "fri b"),
+            )
```

Comment:
```suggestion
            sponsor = EighthSponsor.objects.get_or_create(
                first_name=sponsor_object.first_name, last_name=sponsor_object.last_name, user=sponsor_object
            )[0]
            room = EighthRoom.objects.get_or_create(name=room_num, capacity=capacity)[0]
            activity = EighthActivity.objects.get_or_create(
                name=name,
                description=description,
                default_capacity=capacity,
                presign=(classification == "presign"),
                one_a_day=(classification == "oneaday"),
                sticky=(classification == "sticky"),
                special=(classification == "special"),
                administrative=(classification == "administrative"),
                restricted=(classification == "restricted"),
                wed_a=(date == "wed a"),
                wed_b=(date == "wed b"),
                fri_a=(date == "fri a"),
                fri_b=(date == "fri b"),
            )[0]
```

The `idc` variable is unnecessary.